bluesky_widgets/qt/search_input.py

```diff
@@ -16,54 +26,138 @@ def __init__(self, model, *args, **kwargs):
+        # TODO: check which RButton is selected to update that range
+
     def on_since_view_changed(self, qdatetime):
         # When GUI is updated
         self.model.since = qdatetime.toSecsSinceEpoch()
 
     def on_since_model_changed(self, event):
-        # When model is updated (e.g. from console)
-        qdatetime = QDateTime()
-        qdatetime.setSecsSinceEpoch(event.date)
-        self.since_widget.setDateTime(qdatetime)
+        # When model is updated (e.g. from console or by clicking a QRadioButton)
+        now = time.time()
+        if isinstance(event.date, timedelta):
+            if event.date == timedelta(days=1):
+                self.since_widget.setDateTime(
+                    QDateTime.fromSecsSinceEpoch(now + timedelta(days=-1).total_seconds()))
+                self.until_widget.setDateTime(QDateTime.fromSecsSinceEpoch(now))
+                self.today_widget.setChecked(True)
+            if event.date == timedelta(days=30):
+                self.since_widget.setDateTime(
+                    QDateTime.fromSecsSinceEpoch(now + timedelta(days=-30).total_seconds()))
+                self.until_widget.setDateTime(QDateTime.fromSecsSinceEpoch(now))
+                self.days_widget.setChecked(True)
+            if event.date == timedelta(minutes=60):
+                self.since_widget.setDateTime(
+                    QDateTime.fromSecsSinceEpoch(now + timedelta(minutes=-60).total_seconds()))
+                self.until_widget.setDateTime(QDateTime.fromSecsSinceEpoch(now))
+                self.hour_widget.setChecked(True)
+        else:
+            qdatetime = QDateTime()
+            qdatetime.setSecsSinceEpoch(event.date.timestamp())
+            self.since_widget.setDateTime(qdatetime)
+            self.uncheck_radiobuttons
+            # does a reload make sense?
+            self.model.request_reload()
 
     def on_until_view_changed(self, qdatetime):
         # When GUI is updated
         self.model.until = qdatetime.toSecsSinceEpoch()
 
     def on_until_model_changed(self, event):
         # When model is updated (e.g. from console)
-        qdatetime = QDateTime()
-        qdatetime.setSecsSinceEpoch(event.date)
-        self.until_widget.setDateTime(qdatetime)
+        if not isinstance(event.date, timedelta):
+            qdatetime = QDateTime()
+            qdatetime.setSecsSinceEpoch(event.date.timestamp())
+            self.until_widget.setDateTime(qdatetime)
+            self.uncheck_radiobuttons
+            self.model.request_reload()
+
+    def on_select_today(self):
+        self.model.since = timedelta(days=1)
+        self.model.until = timedelta()
+
+    def on_select_lasthour(self):
+        self.model.since = timedelta(minutes=60)
+        self.model.until = timedelta()
+
+    def on_select_30days(self):
+        self.model.since = timedelta(days=30)
+        self.model.until = timedelta()
+
+    def on_select_all(self):
+        # self.model.until = None
```

Comment:
This should update `self.model` but not touch the UI directly, just like the other `on_select_*` methods. I'm fine with using `0` as the default for now, though I also like the idea of a cheeky choice like Galileo's birthday or something. We can always iterate later. :-D

Reply:
done, see commit